pkg/userdata/helper/kubelet.go

```diff
@@ -131,6 +134,51 @@ func KubeletSystemdUnit(kubeletVersion, cloudProvider, hostname string, dnsIPs [
 	return b.String(), nil
 }
 
+// kubeletConfiguration returns marshaled kubelet.config.k8s.io/v1beta1 KubeletConfiguration
+func kubeletConfiguration(clusterDomain string, clusterDNS []net.IP) (string, error) {
+	var (
+		True  = true
+		False = false
+	)
+	clusterDNSstr := make([]string, 0, len(clusterDNS))
+	for _, ip := range clusterDNS {
+		clusterDNSstr = append(clusterDNSstr, ip.String())
+	}
+
+	cfg := kubeletv1b1.KubeletConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "KubeletConfiguration",
+			APIVersion: "kubelet.config.k8s.io/v1beta1",
+		},
+		Authentication: kubeletv1b1.KubeletAuthentication{
+			X509: kubeletv1b1.KubeletX509Authentication{
+				ClientCAFile: "/etc/kubernetes/pki/ca.crt",
+			},
+			Webhook: kubeletv1b1.KubeletWebhookAuthentication{
+				Enabled: &True,
+			},
+			Anonymous: kubeletv1b1.KubeletAnonymousAuthentication{
+				Enabled: &False,
+			},
+		},
+		Authorization: kubeletv1b1.KubeletAuthorization{
+			Mode: kubeletv1b1.KubeletAuthorizationModeWebhook,
+		},
+		CgroupDriver:          "systemd",
+		ClusterDNS:            clusterDNSstr,
+		ClusterDomain:         clusterDomain,
+		FeatureGates:          map[string]bool{"RotateKubeletServerCertificate": true},
+		ProtectKernelDefaults: true,
+		ReadOnlyPort:          0,
```

Comment:
This field here is going to be omitted which means it won't exist in the `/etc/kubernetes/kubelet.conf`, thus I think we can remove it. In addition, we are not passing any values from the template to it which means it will always be 0.

Reply:
It's more for being explicit / for readability.